docs/hwids.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Hardware IDs
+---
+
+## Introduction
+
+Hardware IDs are used by fwupd to identify specific hardware.
+This is useful as the device-specific identifiers may be the same for different firmware streams.
+Each hardware ID has varying levels of specificity for the hardware, for instance matching only the
+system OEM, or matching up to 8 fields including the system BIOS version.
+
+For instance, Dell and Lenovo might ship a wireless broadband modem with the same chip vendor and
+product IDs of `USB\VID_413C&PID_81D7` and although the two OEMs share the same internal device,
+the firmware may be different.
+To cover this case fwupd allows adding `hardware` requirements that mean we can deploy firmware that
+targets `USB\VID_413C&PID_81D7`, but *only for Dell* or *only for Lenovo* systems.
+
+Microsoft calls these values "CHIDs" and they are generated on Windows using the `ComputerHardwareIds.exe`
+binary which can be found [here](https://learn.microsoft.com/en-us/windows-hardware/drivers/devtest/computerhardwareids).
+The list of Windows 10 CHIDs are:
+
+| HWID            | Contents
+|----------------:|:---------
+| `HardwareID-0`  | Manufacturer + Family + Product Name + SKU Number + BIOS Vendor + BIOS Version + BIOS Major Release + BIOS Minor Release
+| `HardwareID-1`  | Manufacturer + Family + Product Name + BIOS Vendor + BIOS Version + BIOS Major Release + BIOS Minor Release
+| `HardwareID-2`  | Manufacturer + Product Name + BIOS Vendor + BIOS Version + BIOS Major Release + BIOS Minor Release
+| `HardwareID-3`  | Manufacturer + Family + ProductName + SKU Number + Baseboard_Manufacturer + Baseboard_Product
+| `HardwareID-4`  | Manufacturer + Family + ProductName + SKU Number
+| `HardwareID-5`  | Manufacturer + Family + ProductName
+| `HardwareID-6`  | Manufacturer + SKU Number + Baseboard_Manufacturer + Baseboard_Product
+| `HardwareID-7`  | Manufacturer + SKU Number
+| `HardwareID-8`  | Manufacturer + ProductName + Baseboard_Manufacturer + Baseboard_Product
+| `HardwareID-9`  | Manufacturer + ProductName
+| `HardwareID-10` | Manufacturer + Family + Baseboard_Manufacturer + Baseboard_Product
+| `HardwareID-11` | Manufacturer + Family
+| `HardwareID-12` | Manufacturer + Enclosure Type
+| `HardwareID-13` | Manufacturer + Baseboard_Manufacturer + Baseboard_Product
+| `HardwareID-14` | Manufacturer
+
+On Windows, CHIDs are generated from the case sensitive ASCII SMBIOS values, and on Linux the same
```

Comment:
Does Windows actually strip non-visible ASCII characters?
```suggestion
On Windows, CHIDs are generated from the ASCII representation of SMBIOS strings, and on Linux the same
```

Reply:
Nah it's worse than that; Windows just assumes the characters are ASCII. I'll use your suggestion tho.